### PR TITLE
Add justification for why/how the plugin helps

### PR DESCRIPTION
**Pull request by zsims**
Additionally documents checkout vs trigger modes, closing #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,126 @@
...
 
 The plugin has two modes:
   - `mode: trigger` to async trigger another build (of the current pipeline); and
   - `mode: checkout` to checkout the PR merged ref (`refs/pull/123/merge`) rather than the head (`refs/pull/123/head`)
 
-## Example
+# Example
+
+## Checkout Mode
+When using `checkout` mode, the plugin should be specified on each build step (e.g. where a checkout happens)he
```
Comment (tamoore) [inline README.md]: Stray `he`

Comment (zsims) [reply README.md]: Fixed, ta!

File: README.md [hunk 2]
```diff
@@ -1,38 +1,126 @@
...
+    plugins:
       seek-oss/github-merged-pr#v0.0.5:
         mode: trigger
+
+  - label: 'Make something else'
+    commands:
+      - make something-else
 ```
 
 Ensure `Skip pull request builds for existing commits` is set to `false` in your Pipeline settings, as BuildKite will build the branch and skip the PR build.
```
Comment (tamoore) [inline README.md]: Is it worth including a screenshot?

Comment (zsims) [reply README.md]: Yeah good point! Will add one

**Pull request merged**